Realm/RLMSyncConfiguration.mm

```diff
@@ -56,6 +61,50 @@ RLMSyncSystemErrorKind errorKindForSyncError(SyncError error) {
 }
 }
 
+struct BeforeClientResetWrapper {
+    RLMClientResetBeforeBlock block;
+    RLMRealmConfiguration *rlmConfig;
+    void operator()(std::shared_ptr<Realm> local) {
+        RLMSchema *schema;
+        if (rlmConfig.dynamic) {
+            schema = [RLMSchema dynamicSchemaFromObjectStoreSchema:local->schema()];
+        }
+        if (auto cached = RLMGetAnyCachedRealmForPath(rlmConfig.config.path)) { // crash, no rlmConfig, despite being set when realm is open?
```

Comment:
I assume this comment is out of date.

This is missing an `else` and will always overwrite the dynamic schema.